src/Bicep.Core/Parsing/BaseParser.cs

```diff
@@ -817,12 +823,23 @@ private SyntaxBase MemberExpression(ExpressionFlags expressionFlags)
                 {
                     // dot operator
                     Token dot = this.reader.Read();
+                    Token? safeAccessMarker = null;
+                    if (this.Check(TokenType.Question))
+                    {
+                        safeAccessMarker = this.reader.Read();
+                    }
 
                     IdentifierSyntax identifier = this.IdentifierOrSkip(b => b.ExpectedFunctionOrPropertyName());
 
                     if (Check(TokenType.LeftParen))
                     {
                         var functionCall = FunctionCallAccess(identifier, expressionFlags);
+                        if (safeAccessMarker is Token nonNullMarker)
```

Comment:
[nit] Since you're not relying on casting (just doing a null check), you can simplify to either:
```csharp
if (safeAccessMarker is {} nonNullMarker)
```

Or just:
```csharp
if (safeAccessMarker is not null)
```